akka-remote/src/main/java/akka/remote/artery/compress/TopHeavyHittersScanning.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery.compress;
+
+public class TopHeavyHittersScanning {
```

Comment:
Silly reason, was easier to write this: https://github.com/akka/akka/pull/20546/files/9c5ebdbf283862da0d042fabf0c745aec3b0e2d0..11feca1d6b2b2e587397286ef85d457d0eea939b#diff-84014b2f094e6f03567127be21cfc2b8R54
in Scala I'll to for-loop and put a weird index to terminate loop, can do so though if we want to be pure scala here (technically this wastes some bytes of space, but doesnt matter in face of the large arrays inside anyway(